### DISCUSSION TITLE
Compile error with ambigoius MainWindow.InitializeComponent for a new application created with Avalonia .NET Core App (AvaloniaUI) template

### DISCUSSION THREAD

**Discussion by bairog** [AvaloniaUI/Avalonia#12079] [Q&A] [answered]
**Describe the bug**
Cannot compile a new application created with **Avalonia .NET Core App (AvaloniaUI)** template (ver **11.0.0**)

**To Reproduce**

1. Create new project with **Avalonia .NET Core App (AvaloniaUI)** template
2. Update all **Avalonia** packages to latest version (**11.0.0**) version. Also update **XamlNameRefecerceGenerator** to latest version (**1.6.1**)
3. Add reference to **Avalonia.Themes.Fluent** package (**11.0.0**)
4. Compiling results in error:
> Error	CS0111	Type 'MainWindow' already defines a member called 'InitializeComponent' with the same parameter types	AvaloniaApplication	D:\AvaloniaApplication\AvaloniaApplication\Avalonia.NameGenerator\Avalonia.NameGenerator.AvaloniaNameSourceGenerator\AvaloniaApplication.MainWindow.g.cs	19	Active

**Desktop (please complete the following information):**

- OS: Windows 11 21H2 x64
- Version: 11.0.0
- Visual Studio 2022 17.6.4 Pro

**Additional context**
Updating any existing AvaloniaUI application from **0.10.18** to **11.0.0** also produces same compilation error.

**Comment by hez2010** [comment, marked as answer]
You don't need to reference XamlNameRefecerceGenerator in Avalonia 11. It has been made as a build-in feature since Avalonia 11 so referencing XamlNameRefecerceGenerator will result in duplicated generated code.

**Comment by bairog** [comment]
> You don't need to reference XamlNameRefecerceGenerator in Avalonia 11. It has been made as a build-in feature since Avalonia 11 so referencing XamlNameRefecerceGenerator will result in duplicated generated code.

Is that mentioned somewhere? Cannot find it in [Breaking Changes](https://github.com/AvaloniaUI/Avalonia/wiki/Breaking-Changes) docs..

**Comment by hez2010** [comment]
> Cannot find it in Breaking Changes docs..

It is mentioned in the section "XamlNameReferenceGenerator is part of the main package now".

**Reply by bairog** [reply]
Have no idea how could I miss that section.. Sorry for that.

**Comment by timunie** [comment]
@bairog as this is not a bug but a breaking change, let me convert this into a Q&A.

**Reply by bairog** [reply]
No problem. One more question. I used  `<FluentTheme Mode="Light"/>` in **0.10.18**. Now in **11.0.0** I have a compile error **AVLN:0004:**

> Unable to resolve suitable regular or attached property Mode on type Avalonia.Themes.Fluent:Avalonia.Themes.Fluent.FluentTheme

If I use `<FluentTheme />` instead - it compiles. But how to use Light mode in **11.0.0**?

**Reply by maxkatz6** [reply]
Please follow. This page has answers on every your question. 
https://docs.avaloniaui.net/docs/next/stay-up-to-date/upgrade-from-0.10

**Reply by timunie** [reply]
@bairog you are also welcome to join our telegram chat for quick questions. https://t.me/Avalonia
